src/tennisSimulator/MatchSimulator.java

```diff
@@ -1,13 +1,20 @@
 package tennisSimulator;
 
+import java.util.HashMap;
+import java.util.Map;
+
 public class MatchSimulator {
```

Comment:
Overall, this class seems to do very much things (violation of SRP).

I would consider introducing more smaller, singe responsibility classes. Maybe have separate classes that would represent Serve, Game, Set (those are just thoughts, not sure if they will fit here, so consider yourself).